src/typechecker.erl

```diff
@@ -73,6 +73,7 @@
              ,tenv             :: #tenv{}
              ,infer    = false :: boolean()
              ,verbose  = false :: boolean()
+	     ,exhaust  = true  :: boolean()
```

Comment:
Tabs?

Btw, are you sure it should be enabled by default? I think in many cases, we can't handle guards nearly as good as we should, so the following would be considered non-exhaustive:

```Erlang
-spec foo(integer()) -> integer().
foo(N) when is_integer(N) -> N.
```

Reply:
I tried to fix it but I only made things worse. There's some Woodoo going on here. I'll look at it some other day when I'm less tired.

Btw, I avoid doing exhaustiveness checking when there are guards so that shouldn't be a problem. Just think about the fact that all our tests pass with this option on, which made me confident that it can be on by default. But it also probably means that it doesn't actually catch that many cases to begin with.

Reply:
OK, all test cases pass is a good thing (apart from one test case where you added a catch-all clause). I didn't see you skip exhaustiveness checking when there are guards, but that's very good! I think it's a good feature, definitely. Don't get me wrong on that. :-)

The type refinement with guards like the one above is something I have struggled with before so I can probably pick that up again. Both refinement and exhaustiveness would benefit from solving it so the reason for solving it is double.

Reply:
Yeah, improving type refinement with guards is a really nice project!

Again, the whitespace issue seems to be a bug with github.